### PR TITLE
🧪 Added branch level experiment for page Outlink layer

### PR DESCRIPTION
**Pull request by jshamble**
Added branch level experiment for the story ads page outlink layer.
To be used in conjunction with https://critique-ng.corp.google.com/cl/391192225 
to monitor and test the effects of the story ads page outlink before launching to prod.

### CODE REVIEW

File: extensions/amp-story-auto-ads/0.1/story-ad-ui.js [hunk 1]
```diff
@@ -253,12 +254,13 @@ function createCtaLayer_(a, doc, container) {
 
 /**
  * @param {!Document} doc
+ * @param {!Window} win
```
Comment (calebcordry) [inline extensions/amp-story-auto-ads/0.1/story-ad-ui.js:257]: let's just get the window from the doc instead of temporarily adding an param.

File: extensions/amp-story-auto-ads/0.1/test/test-story-ad-ui.js [hunk 2]
```diff
@@ -14,6 +14,15 @@ describes.realWin('story-ad-ui', {amp: true}, (env) => {
   let win;
   let doc;
 
+  /** @const {!{branch: string, control: string, experiment: string}}
+   @visibleForTesting
+  */
+  const AD_PAGE_OUTLINK_EXP = {
```
Comment (calebcordry) [inline extensions/amp-story-auto-ads/0.1/test/test-story-ad-ui.js:20]: Why duplication? Can we just import this?

**Review by calebcordry — CHANGES_REQUESTED** [review]

**Review by calebcordry — APPROVED** [review]
Nice work!

File: build-system/global-configs/prod-config.json [hunk 3]
```diff
@@ -15,6 +15,7 @@
   "disable-a4a-non-sd": 1,
   "amp-cid-backup": 1,
   "story-ad-placements": 0.01,
+  "story-ad-page-outlink": 0,
```
Comment (jridgewell) [inline build-system/global-configs/prod-config.json:18]: Rever the two changes in `canary-config.json` and `prod-config.json`. We're preferring `client-side-experiments-config.json`. See https://github.com/ampproject/amphtml/tree/main/build-system/global-configs#client-side-experiments-configjson

**Note**: The change to `client-side-experiments-config.json` should be done in a separate PR. Changes to the file are propagated to traffic immediately.

**Review by jridgewell — APPROVED** [review]
Owners approval.

**Review by calebcordry — APPROVED** [review]

File: build-system/global-configs/prod-config.json (hunk 3 above)
Comment (jshamble) [reply build-system/global-configs/prod-config.json]: Sounds good, the changes were reverted and I created a new PR Here:
https://github.com/ampproject/amphtml/pull/36030
